### PR TITLE
Scope: Add basic integration tests

### PR DESCRIPTION
**Pull request by ryantxu**
This adds two sample files for scopes and a simple integration test.

```
export KUBECONFIG=$PWD/data/grafana-apiserver/grafana.kubeconfig
```

And then:
```
kubectl apply -f pkg/tests/apis/scopes/testdata/example-scope.yaml --server-side
kubectl apply -f pkg/tests/apis/scopes/testdata/example-scope-dashboard-binding-aaa.yaml --server-side
kubectl apply -f pkg/tests/apis/scopes/testdata/example-scope-dashboard-binding-xyz.yaml --server-side
```

### CODE REVIEW

File: pkg/apis/scope/v0alpha1/register.go
Comment (ryantxu) [inline pkg/apis/scope/v0alpha1/register.go:51]: When we register the internal version, this fails with the error:
```
Double registration of different types for scope.grafana.app/__internal, Kind=WatchEvent: old=k8s.io/apimachinery/pkg/apis/meta/v1.InternalEvent, new=k8s.io/apimachinery/pkg/apis/meta/v1.WatchEvent in scheme "pkg/runtime/scheme.go:100
```

it does not appear to be used, so maybe we just remove it?

**Comment by ryantxu** [comment]
@kylebrandt -- here is a VERY basic test, but at least lets us see what valid examples should look like.  Not implemented in this test is actually calling the filtering -- I think we can merge this, then see if you can add more to it.

@toddtreece -- what are your thought on typed clients?  This uses untyped/dynamic -- ok for a sanity check, but pretty difficult for more complex tests

File: pkg/tests/apis/scopes/scopes_test.go
Comment (ryantxu) [inline pkg/tests/apis/scopes/scopes_test.go:128]: @kylebrandt -- would be good to exercise ListOptions with the fieldSelectors we are implementing

**Comment by kylebrandt** [comment]
I don't think this should still be green after merging in main, since that has the change of `[]Dashboard` to `Dashboard`. So the tests need to be checking for something else I think - in the past when doing this (malformed typed) manually over http api, this would have returned a Warning HTTP header in the response.

**Review by kylebrandt — CHANGES_REQUESTED** [review]
I think we need to see the tests fail (or expect warning?) when there are incorrect specs submitted, right now it seems to pass when it shouldn't.

**Comment by kylebrandt** [comment]
Ah, the `CreationOptions` has options for this:

e.g. `metav1.CreateOptions{FieldValidation: "Strict"}`

```
    /home/kbrandt/go/github.com/grafana/grafana/pkg/tests/apis/scopes/scopes_test.go:123:
        	Error Trace:	/home/kbrandt/go/github.com/grafana/grafana/pkg/tests/apis/scopes/scopes_test.go:123
        	Error:      	Received unexpected error:
        	            	ScopeDashboardBinding in version "v0alpha1" cannot be handled as a ScopeDashboardBinding: strict decoding error: unknown field "spec.dashboards"
        	Test:       	TestIntegrationScopes/Check_create_and_list
```

**Review by kylebrandt — APPROVED** [review]

File: pkg/tests/apis/scopes/scopes_test.go
Comment (toddtreece) [inline pkg/tests/apis/scopes/scopes_test.go:122]: it looks like there's a duplicate error check. is it worth adding a test for the contents of the list?
```suggestion
		require.Len(t, found.Items, 1)
		require.Equal(t,
			"example2",
			mustNestedString(found.Items[0], "metadata", "name"),
		)
```

**Review by toddtreece — APPROVED** [review]

**Pull request merged**
